### PR TITLE
[To merge on 18 May] Adds Get transfer details

### PR DESCRIPTION
**Pull request by natasha-moore-cko**
# Description of changes in PR

[//]: # 'Please add your description here'

## Checklist

- [x] Added a changelog entry to the `changelog.md` file in either `abc_spec` or `nas_spec`
- [ ] Contacted the Tech Docs team about any corresponding guides that need to be updated for www.checkout.com/docs

**Contributing options**: Look at [our documentation](https://checkout.atlassian.net/wiki/spaces/PD/pages/2169506663/API+ref+publication+process) for options to contribute to the API reference.

### CODE REVIEW

File: nas_spec/paths/transfers@{id}.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            $ref: '#/components/schemas/Transfer'
+      headers:
+        Cko-Request-Id:
+          $ref: '#/components/headers/Cko-Request-Id'
+        Cko-Version:
+          $ref: '#/components/headers/Cko-Version'
+    '400':
+      description: Bad Request
+    '401':
+      description: Unauthorized
```
Comment (chris-mannix-cko) [inline nas_spec/paths/transfers@{id}.yaml:32]: Again an issue with our source spec (sorry!), but there should be a 404 here too.

**Comment by chris-mannix-cko** [comment]
Would it be worth updating https://github.com/checkout/checkout-api-reference/blob/master/nas_spec/paths/transfers.yaml
to use the new scope value `transfer:create`?

**Comment by chris-mannix-cko** [comment]
Also, this change brought about some small changes to the existing endpoints too:
*  The ListTransfersResponse shares a schema with the get transfer response - might be worth a) updating it and b) having it use a shared schema so we don't have to define it twice
* A representation of a 'transfer' now always includes a `_self` link, whether that's coming back from create, query, or get

Hopefully you can see the schema changes here: https://github.com/cko-marketplace/transfers-api/pull/87/files

**Comment by natasha-moore-cko** [comment]
> Would it be worth updating https://github.com/checkout/checkout-api-reference/blob/master/nas_spec/paths/transfers.yaml to use the new scope value `transfer:create`?

Sure thing, I was going to create a ticket and do that separately :)

**Review by chris-mannix-cko — DISMISSED** [review]

**Review by chris-mannix-cko — APPROVED** [review]

**Review by ben-ahmady-cko — APPROVED** [review]

**Pull request merged**